### PR TITLE
Upgrading history package

### PR DESCRIPTION
**Pull request by awais786**
Upgrading history package to its latest version. To minimize the django42 scope make this separate PR.

https://django-simple-history.readthedocs.io/en/latest/#section-1
Upgrading the package.
https://github.com/jazzband/django-simple-history/pull/862 ( this pr shows the get_latest_by usage )

Similars PR merged in credentials https://github.com/openedx/credentials/pull/1787
and platform https://github.com/openedx/edx-platform/pull/32880

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -92,20 +92,29 @@ migrate: ## apply database migrations
 html_coverage: ## generate and view HTML coverage report
 	coverage html && open htmlcov/index.html
 
-upgrade: export CUSTOM_COMPILE_COMMAND=make upgrade
-upgrade: piptools ## update the requirements/*.txt files with the latest packages satisfying requirements/*.in
-	# Make sure to compile files after any other files they include!
+COMMON_CONSTRAINTS_TXT=requirements/common_constraints.txt
```
Comment (awais786) [inline Makefile:95]: it was not working as expected. Fixed.

File: enterprise_catalog/apps/catalog/migrations/0037_alter_historicalcontentmetadata_options_and_more.py [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    operations = [
+        migrations.AlterModelOptions(
+            name='historicalcontentmetadata',
+            options={'get_latest_by': ('history_date', 'history_id'), 'ordering': ('-history_date', '-history_id'), 'verbose_name': 'historical Content Metadata', 'verbose_name_plural': 'historical Content Metadata'},
+        ),
+        migrations.AlterModelOptions(
+            name='historicalenterprisecatalog',
+            options={'get_latest_by': ('history_date', 'history_id'), 'ordering': ('-history_date', '-history_id'), 'verbose_name': 'historical Enterprise Catalog', 'verbose_name_plural': 'historical Enterprise Catalogs'},
+        ),
+        migrations.AlterField(
```
Comment (awais786) [inline enterprise_catalog/apps/catalog/migrations/0037_alter_historicalcontentmetadata_options_and_more.py]: This migration 
` model_name='catalogquery',
  name='content_filter',` is part of some old change in this repo.

File: enterprise_catalog/apps/catalog/migrations/0037_alter_historicalcontentmetadata_options_and_more.py [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('catalog', '0036_auto_20230306_1550'),
+    ]
+
+    operations = [
+        migrations.AlterModelOptions(
+            name='historicalcontentmetadata',
+            options={'get_latest_by': ('history_date', 'history_id'), 'ordering': ('-history_date', '-history_id'), 'verbose_name': 'historical Content Metadata', 'verbose_name_plural': 'historical Content Metadata'},
```
Comment (awais786) [inline enterprise_catalog/apps/catalog/migrations/0037_alter_historicalcontentmetadata_options_and_more.py]: its due to history-table.

File: enterprise_catalog/apps/catalog/migrations/0037_auto_20230926_1835.py [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('catalog', '0036_auto_20230306_1550'),
+    ]
+
+    operations = [
+        migrations.AlterModelOptions(
+            name='historicalcontentmetadata',
+            options={'get_latest_by': ('history_date', 'history_id'), 'ordering': ('-history_date', '-history_id'), 'verbose_name': 'historical Content Metadata', 'verbose_name_plural': 'historical Content Metadata'},
```
Comment (awais786) [inline enterprise_catalog/apps/catalog/migrations/0037_auto_20230926_1835.py:18]: these are expected due to history package.

File: enterprise_catalog/apps/catalog/migrations/0037_auto_20230926_1835.py [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+        ),
+        migrations.AlterModelOptions(
+            name='historicalenterprisecatalog',
+            options={'get_latest_by': ('history_date', 'history_id'), 'ordering': ('-history_date', '-history_id'), 'verbose_name': 'historical Enterprise Catalog', 'verbose_name_plural': 'historical Enterprise Catalogs'},
+        ),
+        migrations.AlterField(
+            model_name='catalogquery',
+            name='content_filter',
+            field=jsonfield.fields.JSONField(default=dict, dump_kwargs={'cls': jsonfield.encoder.JSONEncoder, 'ensure_ascii': False, 'indent': 4, 'separators': (',', ':')}, help_text="Query parameters which will be used to filter the discovery service's search/all endpoint results, specified as a JSON object.", load_kwargs={'object_pairs_hook': collections.OrderedDict}),
```
Comment (awais786) [inline enterprise_catalog/apps/catalog/migrations/0037_auto_20230926_1835.py:27]: These are due to some old change in app.

**Comment by UsamaSadiq** [comment]
`django-simple-history` has already been updated in the PR https://github.com/openedx/enterprise-catalog/pull/673 so this is not needed anymore.

**Pull request closed without merging**